src/test/java/seedu/algobase/model/GuiStateTest.java

```diff
@@ -0,0 +1,40 @@
+package seedu.algobase.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.algobase.commons.core.index.Index;
+
+public class GuiStateTest {
+
+    private static final Index OLD_VALUE = Index.fromZeroBased(ModelEnum.PROBLEM.getDisplayTabPaneIndex());
+    private static final Index NEW_VALUE = Index.fromZeroBased(ModelEnum.TAG.getDisplayTabPaneIndex());
+    private static final Index ILLEGAL_VALUE = Index.fromZeroBased(ModelEnum.values().length);
+
+    @Test
+    public void test_setDisplayTabPaneIndex_success() {
```

Comment:
I guess the second part of a test function name should be the condition for the test? It should probably be something like `setDisplayTabPaneIndex_indexWithinRange_success`.